Client/Frontend/Browser/OpenSearch.swift

```diff
@@ -36,13 +44,50 @@ class OpenSearchEngine {
     }
 
     /**
+     * Return the arg that we use for searching for this engine
+     * Problem: the search terms may not be a query arg, they may be part of the URL - how to deal with this?
+     **/
+    private func getQueryArgFromTemplate() -> String? {
+        if let queryStartIndex = searchTemplate.rangeOfString("?")?.startIndex.successor() {
+            let queryArgs = searchTemplate.substringFromIndex(queryStartIndex)
+            let queryParts = queryArgs.componentsSeparatedByString("&").map { $0.componentsSeparatedByString("=") }
+            for part in queryParts {
+                if part[1] == SearchTermComponent {
+                    return part[0]
```

Comment:
This seems really fragile -- won't this crash for URLs that don't have `=` in the query? For example, http://google.com?a&b

Can you use the iOS APIs (`NSURLComponents.queryItems`?) to deconstruct these queries?